lib/encryption/utils/bootstrap.dart

```diff
@@ -292,12 +292,15 @@ class Bootstrap {
         }
         // alright, we re-encrypted all the secrets. We delete the dead weight only *after* we set our key to the default key
       }
-      final updatedAccountData = client.onSync.stream.firstWhere((syncUpdate) =>
-          syncUpdate.accountData != null &&
-          syncUpdate.accountData!.any((accountData) =>
-              accountData.type == EventTypes.SecretStorageDefaultKey));
       await encryption.ssss.setDefaultKeyId(newSsssKey!.keyId);
-      await updatedAccountData;
+      // how useful is this? wouldn't a old ssss be fine here anyway?
+      while (client.accountData
+          .where((_, BasicEvent event) =>
+              event.type == EventTypes.SecretStorageDefaultKey)
+          .isNotEmpty) {
+        Logs().v('Waiting accountData to have m.secret_storage.default_key');
+        await client.oneShotSync(timeout: Duration(seconds: 3));
+      }
```

Comment:
Why do you add the wait logic here, if you already added that to the store function? Also the store should maybe now happen in parallel like the wait did before?

Reply:
because it doesn't use the store function, atleast not for the defaultKeyId? `await encryption.ssss.setDefaultKeyId(newSsssKey!.keyId);`